File: UniShare.API/Models/DTOs/Listings/CreateListingRequest.cs
namespace UniShare.API.Models.DTOs.Listings;

public class CreateListingRequest
{
    public string Title { get; set; } = null!;
    public string Description { get; set; } = null!;
    public Guid CategoryId { get; set; }
    public string ListingType { get; set; } = null!; // "Rent" or "Borrow"
    public Guid? SchoolId { get; set; }
    public Guid? AreaId { get; set; }
    public decimal PricePerDay { get; set; }
    public decimal? DepositAmount { get; set; }
    public string? ConditionNote { get; set; }
    public List<string> TagNames { get; set; } = new();
}

File: UniShare.APP/lib/features/listings/models/create_listing_request.dart
import 'package:json_annotation/json_annotation.dart';
import '../../../core/enums/listing_type.dart';

part 'create_listing_request.g.dart';

@JsonSerializable()
class CreateListingRequest {
  final String title;
  final String description;
  final String categoryId;
  final String? schoolId;
  final String? areaId;
  final ListingType listingType;
  final double pricePerDay;
  final double depositAmount;
  final String? conditionNote;
  final List<String>? tags;

  const CreateListingRequest({
    required this.title,
    required this.description,
    required this.categoryId,
    this.schoolId,
    this.areaId,
    required this.listingType,
    required this.pricePerDay,
    required this.depositAmount,
    this.conditionNote,
    this.tags,
  });

  factory CreateListingRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateListingRequestFromJson(json);

  Map<String, dynamic> toJson() => _$CreateListingRequestToJson(this);
}
